src/main/java/co/ivi/jus/record/review/xuling/SocialSecurityNumber.java

```
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2021, Ling Xu. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ */
+
+package co.ivi.jus.record.review.xuling;
+
+import java.security.MessageDigest;
+import java.util.Arrays;
+
+/**
+ * record class work
+ *
+ * @author jack
+ * @since 2021/11/24
+ **/
+public record SocialSecurityNumber(byte[] ssn) {
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+
+        if (o == null) {
+            return false;
+        }
+
+        if(this.ssn == null) {
+            return false;
+        }
+
+        if (o instanceof SocialSecurityNumber that) {
+            return MessageDigest.isEqual(ssn, that.ssn());
+        }
```

这些代码还可以进一步的简化。 第25行到35行删除掉，逻辑上应该没有什么影响。

---

在逻辑上没有什么问题。这里需要failfast吗?

---

倒不是fail fast，这些代码是冗余的，对于执行结果没有影响。 一般地，从性能和可维护性方面来说，我们不保留、不执行没有用的代码。